src/librustc_mir/transform/qualify_min_const_fn.rs

```diff
@@ -403,28 +403,28 @@ fn check_terminator(
 ///
 /// Adding more intrinsics requires sign-off from @rust-lang/lang.
 fn is_intrinsic_whitelisted(tcx: TyCtxt<'tcx>, def_id: DefId) -> bool {
```

Comment:
Wait, why is there *another* such whitelist now? We shouldn't have two of them. I didn't see this in earlier versions of the patch.
(Please don't amend commits during review, that makes it impossible to see what changed on your side -- because you also did a rebase, GitHub's diff is useless.)

Reply:
Too much stuff changed, it would have been useless to not squash. I basically removed half the implementation because it got deduplicated.

The second whitelist is for `min_const_fn`, so basically which intrinsics are reachable from stable

Reply:
> The second whitelist is for min_const_fn, so basically which intrinsics are reachable from stable

Any chance those could be in the same file? Or maybe even the same function, one that returns both whether the fn is const and whether it is *stable* const?

Also, if this is where stability is decided, then what is the point of the `#[rustc_const_unstable]` you added to `transmute`?

Reply:
> Also, if this is where stability is decided, then what is the point 

I think this is mainly a result of being overly scared of accidental stabilization and thus requiring two levels of activation :laughing:

I'll fix this properly with the `const_stable(since = ...)` attribute. Do you want that to happen in this PR or is it ok if I move all the lists into the same file for now?

Reply:
> I think this is mainly a result of being overly scared of accidental stabilization and thus requiring two levels of activation laughing

Well, we have *four* levels of activation now, spread across 4 different files:
* the implementation
* the first whitelist in `constness.rs`
* the `rustc_const_unstable` attribute
* the second whitelist here

That's what I would call overkill. ;)

> I'll fix this properly with the const_stable(since = ...) attribute. Do you want that to happen in this PR or is it ok if I move all the lists into the same file for now?

No need to do everything in one PR. But I'd prefer to remove the `rustc_const_unstable` from this PR if that is covered by the second whitelist here, and I'd prefer the two whitelists to be merged into one *function* if possible (so there is only one list that classifies intrinsics as non-const, unstable-const, stable-const). Putting them right next to each other in one file is okay I guess if it truly is temporary.

Reply:
> But I'd prefer to remove the rustc_const_unstable from this PR if that is covered by the second whitelist here, 

That attribute is needed not to prevent `transmute` from being used in min const fn, but from being used in `const` items and unstable const fns.

Reply:
That's exactly what I meant, this is a mess. ;)

Why does the attribute not also kick in for "min const fn"?

Reply:
because the whitelist takes precedence over stability attributes -.-

I agree it's a mess, but if I touch it now this PR will grow a lot. It's now in a position where I can do the next PR (const_stable attribute) in a reasonable way without also doing this PR